components/widgets/inputs/users_emails_input.tsx

```diff
@@ -116,7 +116,7 @@ export default class UsersEmailsInput extends React.PureComponent<Props, State>
     }
 
     getOptionValue = (user: UserProfile | EmailInvite): string => {
-        if (user.hasOwnProperty('id')) {
+        if (Object.prototype.hasOwnProperty.call(user, 'id')) {
```

Comment:
Can we use `if (user instanceof UserProfile) {`?

Reply:
This specific change throws a `TS2693: 'UserProfile' only refers to a type, but is being used as a value here.`, but I've improved it [a bit for readability](https://github.com/mattermost/mattermost-webapp/pull/12052/commits/d096f04f4a2ad7fb93c84e50c66756607785780e).